### PR TITLE
CAPZ: add storage-related periodic jobs

### PR DESCRIPTION
**Pull request by chewong**
Setting all the new periodic jobs to 3h to get a faster test signal before changing it to 24h.

/assign @CecileRobertMichon

### CODE REVIEW

File: config/jobs/kubernetes/sig-cloud-provider/azure/sig-azure-config.yaml [hunk 1]
```diff
@@ -294,12 +294,14 @@ periodics:
     - image: gcr.io/k8s-testimages/kubekins-e2e:v20200513-7c0e31a-master
       command:
       - "runner.sh"
-      - "./scripts/ci-conformance.sh"
+      - "./scripts/ci-entrypoint.sh"
```
Comment (CecileRobertMichon) [inline config/jobs/kubernetes/sig-cloud-provider/azure/sig-azure-config.yaml:297]: are there other occurrences of ci-conformance.sh left in sig-cluster-lifecycle jobs? If so, will you change those in a separate PR or should they be changed in this one too?

Comment (chewong) [reply config/jobs/kubernetes/sig-cloud-provider/azure/sig-azure-config.yaml:297]: Good catch, I noticed that `cluster-api-provider-azure-periodics.yaml` is still using `ci-conformance.sh`. I will update this PR. Other than that, I don't think there are any occurrences of `ci-conformance.sh` left in CAPZ-related configs.

File: config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-periodics.yaml [hunk 2]
```diff
@@ -22,7 +22,16 @@ periodics:
...
-          - "./scripts/ci-conformance.sh"
+          - "./scripts/ci-entrypoint.sh"
+        env:
+          - name: FOCUS
+            value: "\\[Conformance\\]|\\[NodeConformance\\]"
+          - name: SKIP
+            value: "\\[Slow\\]|\\[Serial\\]|\\[Flaky\\]"
+          - name: PARALLEL
+            value: "true"
+          - name: USE_CI_ARTIFACTS
```
Comment (CecileRobertMichon) [inline config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-periodics.yaml:33]: this one shouldn't have `USE_CI_ARTIFACTS` set to true, it's meant to test conformance of CAPZ from a known working k8s version (not to test upstream k8s)

Comment (CecileRobertMichon) [reply config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-periodics.yaml:33]: we could set K8s version explicitly here though, I think right now it just defaults to 1.17.5

Comment (chewong) [reply config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-periodics.yaml:33]: I see. I will remove it and let the script use the default value.

**Pull request merged**